Update documentation (#2181)
sonata-project/SonataMediaBundle: docs/reference/security.rst

Built-in security strategy:

* ``sonata.media.security.superadmin_strategy``: DEFAULT - the user needs to have one of the following roles:
  ``ROLE_SUPER_ADMIN`` or ``ROLE_ADMIN`` (although these roles can be configured in SonataAdminBundle)
* ``sonata.media.security.public_strategy``: no restriction, files are public
* ``sonata.media.security.forbidden_strategy``: not possible to retrieve the original file

> * ``sonata.media.security.connected_strategy``: the need to have one of the following roles:

Reviewer: ```suggestion
* ``sonata.media.security.connected_strategy``: the user needs to have one of the following roles:
```

  ``IS_AUTHENTICATED_FULLY`` or ``IS_AUTHENTICATED_REMEMBERED``

> On top of that, there is 3 download modes which can be configured to download the media. The download mode depends on

Reviewer: ```suggestion
On top of that, there are 3 download modes which can be configured to download the media. The download mode depends on
```

the HTTP server you used:

* http: DEFAULT - use php to send the file
* X-Sendfile: use the ``X-Sendfile`` flag (Apache + mod_xsendfile: https://tn123.org/mod_xsendfile/)

> * X-Accel-Redirect: use the ``X-Accel-Redirect`` flag (Nginx: http://wiki.nginx.org/X-accel)

Reviewer: ```suggestion
* X-Accel-Redirect: use the ``X-Accel-Redirect`` flag (Nginx: https://www.nginx.com/resources/wiki/start/topics/examples/x-accel/#x-accel-redirect)
```

Creating your own Security Download Strategy
--------------------------------------------

> The Strategy class must implement the ``DownloadStrategyInterface`` which contains 2 main methods:

Reviewer: ```suggestion
The strategy class must implement the ``DownloadStrategyInterface`` which contains 2 main methods:
```

        }
    }

Let's explain a bit:

> * ``isGranted``: the method test if granted user exists in allowed users for download

Reviewer: ```suggestion
* ``isGranted``: the method test if allowed users contains the authenticated user for download
```
